### PR TITLE
Upgrade JS dependencies and enable image captions capabilities

### PR DESCRIPTION
**Pull request by gsong**
In order to add captions to the markup when converting MDX to HTML, we need to turn on the `showCaptions` option for `gatsby-remark-images` plugin. In order for this to be performant in development, we need to [upgrade the dependencies, specifically `gatsby` and `gatsby-plugin-sharp`](https://www.gatsbyjs.com/docs/reference/release-notes/v2.30/#query-on-demand-and-lazy-images-generally-available).

The `showCaptions` option is turned off for now, until we decide how we want to deal with image alt text, title, and captions.

This PR encompasses the following:

* Upgrade as many dependencies as possible. The only one that couldn't upgraded is `remark-frontmatter` because the latest version requires a newer version of `remark` than what we can use.
* Remove unnecessary prettier configuration.
* Fix corrupt PNGs which causes the build to fail because the new `sharp` library is more strict.
* Fix anonymous default exports which are incompatible with `react-refresh`.
* Fix compound Sass declarations.

See commit messages for more details.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -29,12 +29,11 @@
...
     "bl": "4.0.3",
-    "bootstrap": "4.5",
-    "gatsby": "^2.25.3",
-    "gatsby-background-image": "^0.9.19",
+    "bootstrap": "4.6.0",
+    "gatsby": "^2.31.1",
     "gatsby-cli": "^2.12.93",
     "gatsby-image": "^2.4.1",
-    "gatsby-plugin-algolia": "^0.13.0",
+    "gatsby-plugin-algolia": "^0.16.2",
```
Comment (epbarger) [inline package.json:36]: Seeing this reminded me - this actually needs to be hard-coded to "0.13.1". Releases higher than that have a bug which wipes out your index settings due to a change they made. I started working with them to fix the bug, but the fix hasn't made it in yet

**Review by epbarger — APPROVED** [review]
Approved, just make sure you update the algolia version!

Thanks for getting this done!

File: package.json (hunk 1 above)
Comment (gsong) [reply package.json:36]: @epbarger According to https://www.npmjs.com/package/gatsby-plugin-algolia there's actually no v0.13.1, so I'm going to peg it to 0.13.0, is that OK?

**Comment by cero-miedo** [comment]
Build is taking a long time at `Generating image thumbnails` after `yarn develop`.

We need to optimise the development experience for running locally.

Suggest this PR be blocked for now.

**Comment by gsong** [comment]
@cero-miedo @epbarger I get the opposite results:

| Branch     | No Cache Startup | Cached Startup |
| ---------- | ---------------- | -------------- |
| develop    | 5m30s            | 36s            |
| upg-gatsby | 4m40s            | 37s            |

When you switch branches to test, I would:

1. `rm -rf node_modules`
2. `yarn`
3. `gatsby clean`
4. `gatsby develop`

**Comment by gsong** [comment]
⚠️ This branch has been rebased. You'll want to reset your local branch to mirror upstream.

**Comment by epbarger** [comment]
Tried this out, works great on my machine! @cero-miedo maybe see if George's instructions above get you sorted?

Also, thinking that maybe we don't want to merge this with `showCaptions: true`, and turn that on in a separate PR later. Just because I think we'll probably want to do a least a tiny bit of styling to the captions and I don't think we necessarily need to wait for that to happen to merge this PR

**Comment by gsong** [comment]
@epbarger Agreed we want this merged shortly after the initial release. @cero-miedo Let me know when you've had a chance to look at the captions, and I'll remove captioning from this branch and prepare another branch.

**Comment by gsong** [comment]
⚠️ Rebased.

**Comment by cero-miedo** [comment]
@epbarger @gsong I think this is technically sound, however it introduces some new content problems we probably need to address:

On `ark_gsg/04_using_ark_console/02_ark_clusters_tab/`
image
The component is writing the alt-tag, however in this instance, it contains asterisk characters.

Also, the docs team have assigned their own "Fig" style notation.

**Comment by cero-miedo** [comment]
>Fix anonymous default exports which are incompatible with react-refresh

eslint is compaining about this on `index.js`

**Comment by gsong** [comment]
@cero-miedo @epbarger I've turned off the caption generation on this branch, so this is ready to be merged.

@cero-miedo Feel free to open up another branch (once this is in develop) to work on the caption stuff.

**Comment by gsong** [comment]
⚠️ Rebased.

**Comment by epbarger** [comment]
@gsong I think we're all in agreement that you can merge this now, thanks!

**Pull request merged**